### PR TITLE
containerd: set `LimitNOFILE` on the service

### LINKED ISSUE #1785: containers' file limitation is too low
**containers**:

**What happened**:
When we deploy HD thru our k8s controller, we have a very small limitation of max open files allowed to the process run in the container.
The current limitation allowed is only 1014, which is way too low for processed implied in HDs 

It seems that containerd does not allow to change ulimit values
```
ulimit -n 819200
bash: ulimit: open files: cannot modify limit: Operation not permitted
```

Also, trying to use the operator-sdk API to add additionnal parameters to pod deployment doesn't work because they are _unsafe_

```
diff --git a/pkg/controller/hdservice/hdcontroller/hdcontroller.go b/pkg/controller/hdservice/hdcontroller/hdcontroller.go
index 77adc07..3475843 100644
--- a/pkg/controller/hdservice/hdcontroller/hdcontroller.go
+++ b/pkg/controller/hdservice/hdcontroller/hdcontroller.go
@@ -240,6 +240,9 @@ func NewDeployment(cr *v1alpha1.HdService, ID, cMapName, cType, relName string)
                                                },
                                        },
                                        Volumes: volumes,
+                                       SecurityContext: &corev1.PodSecurityContext{
+                                               Sysctls: []corev1.Sysctl{corev1.Sysctl{Name: "fs.file-max", Value: "100000"}},
+                                       },
                                },
                        },
                },

```
```
kubectl get pod
NAME                                      READY   STATUS             RESTARTS   AGE
default-http-backend-5d5ff5d4f5-8j7v7     1/1     Running            28         8d
....
my-hdservice-monitor-1-7b8968497b-l8dsf   0/1     **SysctlForbidden**    0          1s

```

**What was expected**:

To have a way to change this value, either globally, or directly in the HD operator

### PR DESCRIPTION
**Pull request by NicolasT**
The default 'open files' limit, 1024, is way too low for some of our
applications. Since there's no way to bump this limit from within a
container, and the artificial limit of 1024 file descriptors stems from
an era where FDs were costly, bump the limit to a more reasonable (in
2019) default.

Manual test:

```
[root@bootstrap ~]# kubectl taint node bootstrap node-role.kubernetes.io/bootstrap-
node/bootstrap untainted
[root@bootstrap ~]# kubectl taint node bootstrap node-role.kubernetes.io/infra-
node/bootstrap untainted
[root@bootstrap ~]# kubectl run --restart=Never --image=busybox shell -- sleep 9999
pod/shell created
[root@bootstrap ~]# kubectl exec -ti shell -- sh -c 'ulimit -n'
1048576
```

Fixes: #1785
See: https://github.com/scality/metalk8s/issues/1785
See: https://github.com/containerd/containerd/issues/3201
See: https://github.com/containerd/containerd/pull/3202

### CODE REVIEW

**Comment by NicolasT** [comment]
~Need to test this first.~

**Review by slaperche-scality — DISMISSED** [review]

**Comment by NicolasT** [comment]
I'll approve once @fferrandis signed off in the issue :)

File: salt/metalk8s/container-engine/containerd/files/50-metalk8s.conf
Comment (fferrandis) [inline salt/metalk8s/container-engine/containerd/files/50-metalk8s.conf]: <3

**Review by fferrandis — APPROVED** [review]

**Comment by fferrandis** [comment]
Thank you team !

**Review by slaperche-scality — APPROVED** [review]

**Pull request merged**
